### ISSUE TITLE
Selection is lost when ContextMenu is displayed

### ISSUE THREAD

**Issue by danipen** [AvaloniaUI/Avalonia.Controls.TreeDataGrid#161]
We are displaying the context menu using `Control.ContextRequestedEvent`. Previously, is worked as expected, but since we migrated to preview 6, we're losing the selection when the context menu is being displayed:
 
![rightclick](https://user-images.githubusercontent.com/501613/229776151-68d2e6a8-d54a-4966-81eb-312991483ea7.gif)

**Comment by grokys** [comment]
How are you assigning the context menu? I'm not seeing this behavior in the sample app:

https://user-images.githubusercontent.com/1775141/231992381-8a61e600-7c2e-40a4-bec2-741bcf88e9bd.mp4

**Comment by danipen** [comment]
We're using:

`mTreeDataGrid.ContextRequested += TreeDataGridEventHandlers.ContextRequested;`

And then:

```
void ContextRequested(object sender, ContextRequestedEventArgs e)
{
    if (IsClickedOnHeader(e))
        return;

    e.TryGetPosition(targetControl, out Point p);

    mMenu.Popup(targetControl, p);
}
```

**Comment by grokys** [comment]
Just tried this, and though I'm not sure exactly what you're doing (our `ContextMenu` class doesn't have a `Popup` method, not `MenuFlyout`), I could reproduce with a `MenuFlyout`. The fix was simple though: add `e.Handled = true` to the `ContextRequested` event handler. Does this fix the issue for you?

**Comment by danipen** [comment]
Yes, setting `e.Handled = true` fixes the issue. Just note that in `0.10.x` this was not needed ...

**Comment by grokys** [comment]
The problem comes from this line:

https://github.com/AvaloniaUI/Avalonia.Controls.TreeDataGrid/blob/stable/0.10.x/src/Avalonia.Controls.TreeDataGrid/Selection/TreeDataGridRowSelectionModel.cs#L392

When the pointer is released, `IsRightButtonPressed` is `false` as the right button has been released. Will fix this.

> note that in 0.10.x this was not needed

Strange, I tried the stable 0.10.x branch of `TreeDataGrid` and I see the same behavior there; not sure what could have changed between 0.10.x and 11.x to cause a difference in behavior, but it's academic because the code has a bug.
